src/rgw/rgw_admin.cc

```diff
@@ -3562,6 +3562,16 @@ int main(int argc, const char **argv)
       {
 	RGWZoneGroup zonegroup(zonegroup_id, zonegroup_name);
 	int ret = zonegroup.init(g_ceph_context, store);
+	// handle the special case of default zonegroup
+	if (ret == -ENOENT && zonegroup_name.empty() && zonegroup_id.empty()) {
+	  // check to see there is no realm
+	  RGWRealm realm;
+	  int realm_ret = realm.init(g_ceph_context, store);
```

Comment:
i worry that we'll need this logic in a lot of other places too - is there a reason we need to special-case it here in radosgw-admin, instead of doing something more general in `RGWZoneGroup::read_default_id()`, which already has access to the realm? https://github.com/ceph/ceph/pull/18667 doesn't look quite right, but i think it's on the right track

Reply:
I still hope we will remove the "default" configuration some day ...
back to reality yes it is better doing it in one place. It will be nice to add some tests